Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1410,12 +1410,27 @@ protected function reflectClassMethod($className, MethodReflection $method)
             $this->classesByMethodAnnotations[$annotationClassName][$className][] = $methodName;
         }
 
-        $returnType = $method->getDeclaredReturnType();
-        if ($returnType !== null && !in_array($returnType, ['self', 'null', 'callable', 'void', 'iterable', 'object', 'mixed']) && !TypeHandling::isSimpleType($returnType)) {
-            $returnType = '\\' . $returnType;
-        }
-        if ($method->isDeclaredReturnTypeNullable()) {
-            $returnType = '?' . $returnType;
+        $returnType= $method->getDeclaredReturnType();
+        $applyLeadingSlashIfNeeded = function (string $type): string {
+            if (!in_array($type, ['self', 'null', 'callable', 'void', 'iterable', 'object', 'mixed'])
+                && !TypeHandling::isSimpleType($type)
+                && !TypeHandling::isLiteralType($type)
```

Comment:
The type range covered here is not enough. As per https://www.php.net/manual/en/language.types.type-system.php, `resource`, `never`, `parent` and `static` must also be covered.

Also: I think that L1415 should be encapsuled in `TypeHandling` as well.

Reply:
Makes sense. Never and void still need to be handled separately as they cannot be used in union types. 

Also I just noticed that union types can contain intersection types in brackets … but it would be fair to ignore this in a bugfix for Neos 7.3